src/server/game/Battlegrounds/BattlegroundMgr.cpp

```diff
@@ -514,50 +543,10 @@ bool BattlegroundMgr::CreateBattleground(CreateBattlegroundData& data)
 {
     // Create the BG
     Battleground* bg = NULL;
-    switch (data.bgTypeId)
-    {
-        case BATTLEGROUND_AV:
-            bg = new BattlegroundAV;
-            break;
-        case BATTLEGROUND_WS:
-            bg = new BattlegroundWS;
-            break;
-        case BATTLEGROUND_AB:
-            bg = new BattlegroundAB;
-            break;
-        case BATTLEGROUND_NA:
-            bg = new BattlegroundNA;
-            break;
-        case BATTLEGROUND_BE:
-            bg = new BattlegroundBE;
-            break;
-        case BATTLEGROUND_EY:
-            bg = new BattlegroundEY;
-            break;
-        case BATTLEGROUND_RL:
-            bg = new BattlegroundRL;
-            break;
-        case BATTLEGROUND_SA:
-            bg = new BattlegroundSA;
-            break;
-        case BATTLEGROUND_DS:
-            bg = new BattlegroundDS;
-            break;
-        case BATTLEGROUND_RV:
-            bg = new BattlegroundRV;
-            break;
-        case BATTLEGROUND_IC:
-            bg = new BattlegroundIC;
-            break;
-        case BATTLEGROUND_AA:
-            bg = new Battleground;
-            break;
-        case BATTLEGROUND_RB:
-            bg = new Battleground;
-            break;
-        default:
-            return false;
-    }
+    bg = bgtypeToBattleground[data.bgTypeId];
+
+    if (bg == NULL)
```

Comment:
nullptr?

Reply:
if in the hashmap you give a wrong key, in this case it should return NULL as value, so I made a check before using the variable bg.
I always wrote these piece of code inspiring by the original code.
In the switch case statement there is a default [return value "NULL"](https://github.com/azerothcore/azerothcore-wotlk/pull/1444/commits/7fde499e7e0e255447f2cf32181c87dc0e39dd67#diff-85ba4e80476ac43de064d54592507596L477)  so I made it.